errors/errors.go

```diff
@@ -0,0 +1,22 @@
+package errors
+
+import (
+	"encoding/hex"
+	"fmt"
+)
+
+// NoSuchObject is an error type that occurs when no object with a given object
+// ID is available.
+type NoSuchObject struct {
+	Oid []byte
+}
+
+// Error implements the error.Error() function.
+func (e *NoSuchObject) Error() string {
+	return fmt.Sprintf("gitobj: no such object: %s", hex.EncodeToString(e.Oid))
+}
+
+func IsNoSuchObject(e error) bool {
+	_, ok := e.(*NoSuchObject)
+	return ok
```

Comment:
One thing we might consider is also checking that `e != nil`, since `(*NoSuchObject)(nil)` (read: "cast `nil` to type `*NoSuchObject`)" is a valid thing to accept in this function.

Reply:
Due to nils of different types comparing unequal, I would have to do something like this:

```go
	err, ok := e.(*noSuchObject)
	return ok && err != nil
```

I've squashed that in.